wrappers/net/grpc/client_interceptors.go

```diff
@@ -0,0 +1,62 @@
+package epsagongrpc
+
+import (
+	"context"
+	"github.com/epsagon/epsagon-go/epsagon"
+	"github.com/epsagon/epsagon-go/protocol"
+	"github.com/epsagon/epsagon-go/tracer"
+	jsoniter "github.com/json-iterator/go"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"strconv"
+)
+
+const EPSAGON_TRACEID_HEADER_KEY = "epsagon-trace-id"
+
+func InjectEpsagonTracerContextID(ctx context.Context, event *protocol.Event) context.Context {
+	traceID := epsagon.GenerateEpsagonTraceID()
+
+	addTraceIdToEvent(traceID, event)
+
+	return metadata.AppendToOutgoingContext(ctx, EPSAGON_TRACEID_HEADER_KEY, traceID)
+}
+
+func addTraceIdToEvent(traceID string, event *protocol.Event) {
+	event.Resource.Metadata[tracer.EpsagonGRPCTraceIDKey] = traceID
+}
+
+
+// UnaryClientInterceptor returns a new unary server interceptor for OpenTracing.
+func UnaryClientInterceptor(args ...context.Context) grpc.UnaryClientInterceptor {
+
+	wrapperTracer := epsagon.ExtractTracer(args)
```

Comment:
So I think the client doesn't need this logic - it can extract the tracer from the interceptor func I guess? This way, the client  can be used from multiple goroutines and be traced. Let's talk about it